### PR TITLE
fixed from version boundaries using pack metadata

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31389

## Description
fixed the version boundaries of relevant demisto instances using pack metadata

@IkaDemisto FYI

### CODE REVIEW

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -1135,7 +1142,9 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
     return tests, packs_to_install
 
 
-def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id_set: dict) -> Tuple[str, str]:
+def get_from_version_and_to_version_bounderies(all_modified_files_paths: set,
+                                               id_set: dict,
+                                               modified_metadata_list: set = None) -> Tuple[str, str]:
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1147]:
```suggestion
                                               modified_metadata_list: set = set([])) -> Tuple[str, str]:
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 2]
```diff
@@ -1147,9 +1156,22 @@ def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id
         (string, string). The boundaries of the lowest from version (defaults to 0.0.0)
          and highest to version (defaults to 99.99.99)
     """
+    modified_metadata_list = modified_metadata_list if modified_metadata_list else set([])
     max_to_version = LooseVersion('0.0.0')
     min_from_version = LooseVersion('99.99.99')
     max_from_version = LooseVersion('0.0.0')
+
+    for metadata in modified_metadata_list:
+        pack_metadata_path = os.path.join(tools.pack_name_to_path(metadata), PACKS_PACK_META_FILE_NAME)
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1165]: Isn't `tools.pack_name_to_path(metadata)` return the path already?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -1071,6 +1074,10 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
     if modified_files_with_relevant_tests:
         tests, packs_to_install = find_tests_and_content_packs_for_modified_files(modified_files_with_relevant_tests,
                                                                                   conf, id_set)
+    for pack in modified_metadata_list:
+        pack_tests = get_tests_for_pack(tools.pack_name_to_path(pack))
+        packs_to_install.add(pack)
+        tests = tests.union(pack_tests)
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1080]:
```suggestion

```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 4]
```diff
@@ -1147,9 +1156,22 @@ def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id
         (string, string). The boundaries of the lowest from version (defaults to 0.0.0)
          and highest to version (defaults to 99.99.99)
     """
+    modified_metadata_list = modified_metadata_list if modified_metadata_list else set([])
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1159]:
```suggestion
```

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 2 above)
Comment (yaakovi) [reply Tests/scripts/collect_tests_and_content_packs.py:1165]: I'll update the variable names, it contains only the pack name.
I kept the original name - that was a mistake...

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 1 above)
Comment (yaakovi) [reply Tests/scripts/collect_tests_and_content_packs.py:1147]: no, that will be a bug

image

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 4 above)
Comment (yaakovi) [reply Tests/scripts/collect_tests_and_content_packs.py:1159]: see previous comment

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 3 above)
Comment (yaakovi) [reply Tests/scripts/collect_tests_and_content_packs.py:1080]: not sure why that came from, will remove

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 1 above)
Comment (ShahafBenYakir) [reply Tests/scripts/collect_tests_and_content_packs.py]: :yaakovi:

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**